Reformat code with IntelliJ auto-reformat function.

Thomas--S opened the pull request:
No code was changed by hand.

I don't know if this is wanted.
Feel free to close it.

VanessaE commented on concrete/init.lua:
```diff
@@ -8,12 +8,12 @@ technic.concrete_posts = {}
 for i = 0, 31 do
-	minetest.register_alias("technic:concrete_post"..i,
-			"technic:concrete_post")
+	minetest.register_alias("technic:concrete_post" .. i,
```
Don't put spaces around a ".." concat operator.

Thomas--S commented:
Thanks for your comments, however, I these changes will have a very low priority for me as they are quite time-consuming for not so much use. If someone wants to help, I can allow them to push to this branch.

kaeza commented on technic/machines/HV/forcefield.lua:
```diff
@@ -43,40 +43,40 @@ local function update_forcefield(pos, meta, active, first)
 	local MinEdge, MaxEdge = vm:read_from_map(vector.subtract(pos, range),
-			vector.add(pos, range))
-	local area = VoxelArea:new({MinEdge = MinEdge, MaxEdge = MaxEdge})
+		vector.add(pos, range))
```
Can you join this line with the above one? I know you're just reformatting it, but It's confusing.

kaeza commented on technic/machines/HV/forcefield.lua:
```diff
@@ -214,9 +214,13 @@ minetest.register_node("technic:forcefield_emitter_off", {
-	groups = {cracky = 1, technic_machine = 1, technic_hv = 1,
-			not_in_creative_inventory=1},
+	tiles = { "technic_forcefield_emitter_on.png" },
+	groups = {
```
It's fine here as the line would be too long. It's more readable this way.

kaeza commented on technic/machines/HV/forcefield.lua:
```diff
@@ -236,27 +240,29 @@ minetest.register_node("technic:forcefield", {
-			length = 1.0,
-		},
-	}},
+	tiles = {
```
I prefer the old format here. It can be changed later if we need to specify anything else.

kaeza commented on technic/machines/HV/generator.lua:
```diff
@@ -3,11 +3,11 @@ minetest.register_alias("hv_generator", "technic:hv_generator")
-		{'technic:carbon_plate',          'technic:mv_generator',   'technic:composite_plate'},
-		{'pipeworks:tube_1',              'technic:hv_transformer', 'pipeworks:tube_1'},
-		{'technic:stainless_steel_ingot', 'technic:hv_cable',       'technic:stainless_steel_ingot'},
+		{ 'technic:carbon_plate', 'technic:mv_generator', 'technic:composite_plate' },
```
These are more readable if aligned IMHO.

kaeza commented on technic/machines/HV/nuclear_reactor.lua:
```diff
@@ -22,37 +22,37 @@ local reactor_desc = S("@1 Nuclear Reactor Core", S("HV")),
-	"list[current_name;src;2,1;3,2;]"..
-	"list[current_player;main;0,5;8,4;]"..
-	"listring[]"
+"invsize[8,9;]" ..
```
Uh, why is this reformatted to the beginning of the line? It should either be directly after the equals sign, or indented by two tabs like the rest.

kaeza commented:
I began to take a look at it, but it's unreviewable.

I'm sorry, but I don't think this should go in as-is. It touches too much stuff and while it makes it more readable in some cases, those cases are few and far between. It also makes other stuff be less readable (like recipes). I'd personally suggest doing this manually and incrementally only fixing the most egregious stuff.

VanessaE commented:
I skimmed through the entire changeset (holy crap!), my 6 comments cover all of the categories of stuff I thought needed work.

That said, I think this would be better split into multiple commits, perhaps one per changed file.

DS-Minetest commented on technic/machines/HV/quarry.lua:
```diff
@@ -109,32 +106,32 @@ local function quarry_run(pos, node)
-			(pdir.x == -1 and vector.new(0,0,1) or
-			vector.new(1,0,0)))
+		local qdir = pdir.x == 1 and vector.new(0, 0, -1) or
+				(pdir.z == -1 and vector.new(-1, 0, 0) or
```
On minetest dev wiki it makes the `and` and `or` at line ends: http://dev.minetest.net/Lua_code_style_guidelines#Continuation_lines

VanessaE replied:
That's for engine and minetest_game and the like, not for mods.  We don't have an official guideline for Technic, rather it's personal preference.

DS-Minetest commented on concrete/init.lua:
```diff
@@ -26,43 +26,43 @@ end
-		{'','', steel_ingot},
-		{'',steel_ingot,''},
-		{steel_ingot, '', ''},
+		{ '', '', steel_ingot },
```
http://dev.minetest.net/Lua_code_style_guidelines#Spaces

Thomas--S commented:
I think I addressed most of the comments now. I don't plan to align tables, as this also makes adding new entries or deleting old ones harder. Also, I don't plan to do the changes where I would have to go through all the files to manually fix the issues.

VanessaE commented:
That commit that removed spacing around braces wasn't needed; rather having a space between braces and their contents makes it more readable.  Suggest reverting that commit (or just another run through Intellij to re-un-fix them), and then squash it all.

VanessaE commented:
As for aligning tables and other manual stuff, that can be done some time later, probably by hand as needed.

Thomas--S commented:
Could you maybe check the style again, before I finally squash?

VanessaE commented:
lgtm from a quick glance.  Squash it and I'll review again later.  @kaeza I know you don't like huge changesets, but can you spare 12 or so hours to go through it also? :smile:

DS-Minetest commented on concrete/init.lua:
```diff
@@ -9,11 +9,11 @@ local S = rawget(_G, "intllib") and intllib.Getter() or function(s) return s end
 for i = 0, 31 do
 	minetest.register_alias("technic:concrete_post"..i,
-			"technic:concrete_post")
+		"technic:concrete_post")
```
http://dev.minetest.net/Lua_code_style_guidelines#Continuation_lines
Does this PR mostly make changes against the lua code style guidelines? :confused: 
Minetest's lua code style guidelines is not that bad, also many people might be habituated to it.
=> I see this all as step into the false direction. If this is merged it can hardly be undone because people are too lazy to change that much and too scared to write something that doesn't fit to the rest. :-1:

Thomas--S replied:
I use mostly the IntelliJ standard reformat settings here. They seem to be somewhat accepted in general. If this doesn't apply for this project, I have no problem with this PR being closed. I only wanted to introduce a consistent code style (although this may not be necessary here).

VanessaE replied:
It's fine for a project to have a consistent coding style, but as I said, I'm pretty sure that the Lua code style guidelines aren't directed at mods, they're directed at stuff that goes into mt_game or the various bits that come with the engine.  And even if they are, the writers of that page have no place dictating what an independent project like Technic does.

Code style is about readability, and in a Minetest mod, a number of those suggestions (such as no spaces between a table's contents and its curly braces) make for less-readable code.

DS-Minetest replied:
If this PR will really be merged, technic would need to set clear code style guidelines. Else everyone will be allowed to apply his own code style and everything gets mixed up and there's totally chaos.
=> So, my opinion: **First make clear code style guidelines**, then make this PR accordingly to it.

Thomas--S replied:
This here should lead to the code style used in this PR:

1. IntelliJ community edition with LUA plugin: Reformat whole project
2. Replace in whole project:
    - ` .. ` :arrow_right: `..` _Removes spaces from `..`_
    - ` ..$` :arrow_right: `..` (Regex) _Removes spaces from `..` at line ends_
    - `'([a-zA-Z:_1-9-]*)'` :arrow_right:  `"$1"` (Regex executed on `*.lua`) _Replaces `'` with `"`_
    - `([-0-9]{1,2})/([0-9]{1,2})([ ]?[,\}])?` :arrow_right: `$1/$2$3` (Regex) _Removes spaces from `/` in e.g. nodebox definitions_

DS-Minetest commented on concrete/init.lua:
```diff
@@ -73,49 +73,49 @@ minetest.register_craftitem(":technic:rebar", {
+local box_front = { -0.1, -0.3, 0, 0.1, 0.3, -0.5 }
+local box_back = { -0.1, -0.3, 0, 0.1, 0.3, 0.5 }
+local box_left = { 0, -0.3, -0.1, -0.5, 0.3, 0.1 }
+local box_right = { 0, -0.3, -0.1, 0.5, 0.3, 0.1 }
```
Those spaces were good.

This whole automatic thing doesn't seem to be that good. Code style by people is much better readable. :-1:
Some small changes are ok like removing spaces at the ends (but this is also done by many code editing programs automatically on saving).

Thomas--S commented:
I'm closing this now becaue of three reasons:
- This seems to be controversial
- It has quite many merge conflicts
- It is a low-priority change
